module/VuFind/src/VuFind/Db/Table/Search.php

```diff
@@ -290,6 +290,7 @@ public function saveSearch(
                 $minified = $normalized->getMinified();
                 $minified->id = $existingRow->getSearchObject()->id;
                 $existingRow->search_object = serialize($minified);
+                $existingRow->session_id = $sessionId;
```

Comment:
Is this a bug fix that should be committed separately? I'm not sure I understand the need for it, though -- if it's a saved search, it should have been retrieved by user ID, and if it's part of the current session, the session ID should already be set. Am I missing some other factor?

Reply:
It's not a bug, or that depends on how you take it. I think it should reflect the last session it was executed in, but it not being updated doesn't cause any errors in current code base. 

But if we keep it up to date, we can always retrieve last searches by session id + search id, and won't need to carry user id around.